src/core/stream/orchestrator/stream_orchestrator.ts

```diff
@@ -255,14 +257,34 @@ export default function StreamOrchestrator(
      * @returns {boolean}
      */
     function isOutOfPeriodList(time : number) : boolean {
-      const head = periodList.head();
-      const last = periodList.last();
-      if (head == null || last == null) { // if no period
+      const periodListLength = periodList.length();
+      if (periodListLength === 0) {
         return true;
       }
-      return head.start > time ||
-            (last.end == null ? Infinity :
-                                last.end) < time;
+      for (let i = 0; i < periodListLength; i++) {
```

Comment:
I don't think we want it here, no? This function is here to check if the given time is out of the boundaries of the given Period, I think to know when new `PeriodBuffers` need to be created.

I don't think the true content start and end is important here.